File: Python/StringConstruction/string_construction.py
#!/bin/python
# https://www.hackerrank.com/challenges/string-construction

import sys

def stringConstruction(s):
    # Complete this function
    p = ""
    cost = 0

    for i in s:
        if i not in p:
            cost += 1
        p += i

    return cost

q = int(raw_input().strip())
for a0 in xrange(q):
    s = raw_input().strip()
    result = stringConstruction(s)
    print(result)

File: Ruby/StringConstruction/string_construction.rb
#!/bin/ruby

def stringConstruction(s)
    # Complete this function
    p_str = ""
    cost = 0

    s.split("").each do |i|
        if !p_str.include? i
           cost += 1
        end
        p_str += i
    end

    return cost
end

q = gets.strip.to_i
for a0 in (0..q-1)
    s = gets.strip
    result = stringConstruction(s)
    puts result;
end
